Incompatible BouncyCastle  FIPS changes in bookkeeper 4.14

**Describe the bug**
In [bookkeeper # 2631](https://github.com/apache/bookkeeper/pull/2631), the default BouncyCastle was changed from non-fips into fips version. But the default version of BouncyCastle in Pulsar is the [non-fips](https://github.com/apache/pulsar/blob/v2.8.0/pulsar-client/pom.xml#L56) one(aimed to make it compatible with the old version of Pulsar). 

Bouncy Castle provides both FIPS and non-FIPS version, but in a JVM, it can not include both of the 2 versions(non-Fips and Fips), and we have to exclude the current version before including the other. This make the backward compatible a little hard, and that's why Pulsar has to involve individual module for [Bouncy Castle](https://pulsar.apache.org/docs/en/security-bouncy-castle).

Pulsar excluded the dependencies of [bookkeeper-server's BouncyCastle](https://github.com/eolivelli/pulsar/blob/2de612791f0334fe469e8e0a58e291358af77cdb/pom.xml#L303) in Pulsar's pom file, and Pulsar only includes the non-fips version, but the bookkeeper-server still wants to use the hard-coded [fips version](https://github.com/Ghatage/bookkeeper/blob/fe45f8f52ba24706ba9b9226b73953b93ef3cefa/bookkeeper-server/src/main/java/org/apache/bookkeeper/tls/TLSContextFactory.java#L63) in [bookkeeper # 2631](https://github.com/apache/bookkeeper/pull/2631)

And if we want to start BookKeeper with TLS enabled through Pulsar's binary, it will meet the following error:
```
Exception in thread "main" java.lang.NoClassDefFoundError: org/bouncycastle/jcajce/provider/BouncyCastleFipsProvider
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:315)
	at org.apache.bookkeeper.common.util.ReflectionUtils.forName(ReflectionUtils.java:49)
	at org.apache.bookkeeper.tls.SecurityProviderFactoryFactory.getSecurityProviderFactory(SecurityProviderFactoryFactory.java:39)
	at org.apache.bookkeeper.proto.BookieServer.<init>(BookieServer.java:129)
	at org.apache.bookkeeper.server.service.BookieService.<init>(BookieService.java:52)
	at org.apache.bookkeeper.server.Main.buildBookieServer(Main.java:304)
	at org.apache.bookkeeper.server.Main.doMain(Main.java:226)
	at org.apache.bookkeeper.server.Main.main(Main.java:208)
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 9 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. using pulsar 2.8.0,
2. start bookkeeper through `bin/pulsar bookie`, with [TLS enabled](https://bookkeeper.apache.org/docs/latest/security/tls/).

**Expected behavior**
By using `bin/pulsar bookie`, BookKeeper server should able to start success.
BK should not hard-coded fips version for Bouncy Castle, it should have an option to use non-fips version to make it compatible.

**Additional context**
We may need to provide the fix in the BookKeeper side first, and then change the dependency of bookkeeper version in Pulsar.

we should also add a test in pulsar, which enable tls for bookkeeper and use bc-non-fips, to cover this error.

```
14:06:44.561 [BookKeeperClientWorker-OrderedExecutor-0-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught
java.lang.NoClassDefFoundError: org/bouncycastle/jcajce/provider/BouncyCastleFipsProvider
at java.lang.Class.forName0(Native Method) ~[?:?]
at java.lang.Class.forName(Class.java:315) ~[?:?]
at org.apache.bookkeeper.common.util.ReflectionUtils.forName(ReflectionUtils.java:49) ~[org.apache.bookkeeper-bookkeeper-common-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.tls.SecurityProviderFactoryFactory.getSecurityProviderFactory(SecurityProviderFactoryFactory.java:39) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.proto.DefaultPerChannelBookieClientPool.<init>(DefaultPerChannelBookieClientPool.java:67) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.proto.BookieClientImpl.lookupClient(BookieClientImpl.java:211) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.proto.BookieClientImpl.addEntry(BookieClientImpl.java:318) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[org.apache.bookkeeper-bookkeeper-server-4.14.1.jar:4.14.1]
at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [org.apache.bookkeeper-bookkeeper-common-4.14.1.jar:4.14.1]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty-netty-common-4.1.63.Final.jar:4.1.63.Final]
at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider
at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[?:?]
at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
... 14 more
14:06:46.566 [pulsar-web-40-3]
```

--- a/distribution/server/src/assemble/LICENSE.bin.txt
+++ b/distribution/server/src/assemble/LICENSE.bin.txt
@@ -446,7 +446,7 @@ The Apache Software License, Version 2.0
  * SnakeYaml -- org.yaml-snakeyaml-1.27.jar
  * RocksDB - org.rocksdb-rocksdbjni-6.10.2.jar
  * Google Error Prone Annotations - com.google.errorprone-error_prone_annotations-2.5.1.jar
- * Apache Thrifth - org.apache.thrift-libthrift-0.12.0.jar
+ * Apache Thrifth - org.apache.thrift-libthrift-0.14.2.jar
  * OkHttp3
      - com.squareup.okhttp3-logging-interceptor-3.14.9.jar
      - com.squareup.okhttp3-okhttp-3.14.9.jar

--- a/distribution/server/src/assemble/LICENSE.bin.txt
+++ b/distribution/server/src/assemble/LICENSE.bin.txt
@@ -390,31 +390,31 @@ The Apache Software License, Version 2.0
     - org.apache.logging.log4j-log4j-1.2-api-2.14.0.jar
  * Java Native Access JNA -- net.java.dev.jna-jna-4.2.0.jar
  * BookKeeper
-    - org.apache.bookkeeper-bookkeeper-common-4.14.1.jar
-    - org.apache.bookkeeper-bookkeeper-common-allocator-4.14.1.jar
-    - org.apache.bookkeeper-bookkeeper-proto-4.14.1.jar
-    - org.apache.bookkeeper-bookkeeper-server-4.14.1.jar
-    - org.apache.bookkeeper-bookkeeper-tools-framework-4.14.1.jar
-    - org.apache.bookkeeper-circe-checksum-4.14.1.jar
-    - org.apache.bookkeeper-cpu-affinity-4.14.1.jar
-    - org.apache.bookkeeper-statelib-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-api-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-common-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-java-client-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-java-client-base-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-proto-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-server-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-service-api-4.14.1.jar
-    - org.apache.bookkeeper-stream-storage-service-impl-4.14.1.jar
-    - org.apache.bookkeeper.http-http-server-4.14.1.jar
-    - org.apache.bookkeeper.http-vertx-http-server-4.14.1.jar
-    - org.apache.bookkeeper.stats-bookkeeper-stats-api-4.14.1.jar
-    - org.apache.bookkeeper.stats-prometheus-metrics-provider-4.14.1.jar
-    - org.apache.distributedlog-distributedlog-common-4.14.1.jar
-    - org.apache.distributedlog-distributedlog-core-4.14.1-tests.jar
-    - org.apache.distributedlog-distributedlog-core-4.14.1.jar
-    - org.apache.distributedlog-distributedlog-protocol-4.14.1.jar
-    - org.apache.bookkeeper.stats-codahale-metrics-provider-4.14.1.jar
+    - org.apache.bookkeeper-bookkeeper-common-4.14.2.jar
+    - org.apache.bookkeeper-bookkeeper-common-allocator-4.14.2.jar
+    - org.apache.bookkeeper-bookkeeper-proto-4.14.2.jar
+    - org.apache.bookkeeper-bookkeeper-server-4.14.2.jar
+    - org.apache.bookkeeper-bookkeeper-tools-framework-4.14.2.jar
+    - org.apache.bookkeeper-circe-checksum-4.14.2.jar
+    - org.apache.bookkeeper-cpu-affinity-4.14.2.jar
+    - org.apache.bookkeeper-statelib-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-api-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-common-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-java-client-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-java-client-base-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-proto-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-server-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-service-api-4.14.2.jar
+    - org.apache.bookkeeper-stream-storage-service-impl-4.14.2.jar
+    - org.apache.bookkeeper.http-http-server-4.14.2.jar
+    - org.apache.bookkeeper.http-vertx-http-server-4.14.2.jar
+    - org.apache.bookkeeper.stats-bookkeeper-stats-api-4.14.2.jar
+    - org.apache.bookkeeper.stats-prometheus-metrics-provider-4.14.2.jar
+    - org.apache.distributedlog-distributedlog-common-4.14.2.jar
+    - org.apache.distributedlog-distributedlog-core-4.14.2-tests.jar
+    - org.apache.distributedlog-distributedlog-core-4.14.2.jar
+    - org.apache.distributedlog-distributedlog-protocol-4.14.2.jar
+    - org.apache.bookkeeper.stats-codahale-metrics-provider-4.14.2.jar
   * Apache HTTP Client
     - org.apache.httpcomponents-httpclient-4.5.13.jar
     - org.apache.httpcomponents-httpcore-4.4.13.jar

--- a/pom.xml
+++ b/pom.xml
@@ -103,7 +103,7 @@ flexible messaging model and an intuitive client API.</description>
     <!-- apache commons -->
     <commons-compress.version>1.21</commons-compress.version>
 
-    <bookkeeper.version>4.14.1</bookkeeper.version>
+    <bookkeeper.version>4.14.2</bookkeeper.version>
     <zookeeper.version>3.6.3</zookeeper.version>
     <snappy.version>1.1.7</snappy.version> <!-- ZooKeeper server -->
     <dropwizardmetrics.version>3.2.5</dropwizardmetrics.version> <!-- ZooKeeper server -->

--- a/pulsar-sql/presto-distribution/LICENSE
+++ b/pulsar-sql/presto-distribution/LICENSE
@@ -414,18 +414,18 @@ The Apache Software License, Version 2.0
     - async-http-client-2.12.1.jar
     - async-http-client-netty-utils-2.12.1.jar
   * Apache Bookkeeper
-    - bookkeeper-common-4.14.1.jar
-    - bookkeeper-common-allocator-4.14.1.jar
-    - bookkeeper-proto-4.14.1.jar
-    - bookkeeper-server-4.14.1.jar
-    - bookkeeper-stats-api-4.14.1.jar
-    - bookkeeper-tools-framework-4.14.1.jar
-    - circe-checksum-4.14.1.jar
-    - codahale-metrics-provider-4.14.1.jar
-    - cpu-affinity-4.14.1.jar
-    - http-server-4.14.1.jar
-    - prometheus-metrics-provider-4.14.1.jar
-    - codahale-metrics-provider-4.14.1.jar
+    - bookkeeper-common-4.14.2.jar
+    - bookkeeper-common-allocator-4.14.2.jar
+    - bookkeeper-proto-4.14.2.jar
+    - bookkeeper-server-4.14.2.jar
+    - bookkeeper-stats-api-4.14.2.jar
+    - bookkeeper-tools-framework-4.14.2.jar
+    - circe-checksum-4.14.2.jar
+    - codahale-metrics-provider-4.14.2.jar
+    - cpu-affinity-4.14.2.jar
+    - http-server-4.14.2.jar
+    - prometheus-metrics-provider-4.14.2.jar
+    - codahale-metrics-provider-4.14.2.jar
   * Apache Commons
     - commons-cli-1.2.jar
     - commons-codec-1.15.jar

